gazebo_ros/src/node.cpp

```diff
@@ -98,13 +103,35 @@ Node::SharedPtr Node::Get(sdf::ElementPtr sdf)
     }
   }
 
+  // set full node name
+  if (ns[0] == '/' && ns.size() == 1) {
+    full_name = ns + name;
+  } else {
+    full_name = ns + "/" + name;
+  }
+
+  // check if node with the same name exists already
+  if (static_existing_nodes_.check_node(full_name)) {
+    RCLCPP_ERROR(
+      internal_logger(),
+      "Found multiple nodes with same name: %s. This might be due to multiple plugins using same "
+      "name to solve this either change one of the the plugin names or use a different namespace. "
+      "The error might also result from a custom plugin inheriting from one of the GazeboRosPlugin "
+      "this can be solved by accessing node object of the parent class itself instead of creating "
+      "a new node object for custom plugin.", full_name.c_str());
+    return nullptr;  // this makes the gazebo shutdown safely
```

Comment:
I guess this is okay for plugins that check for `nullptr`. We should document this new behavior in the docs for this function.

Reply:
Where should I document this.

Reply:
Here: https://github.com/ros-simulation/gazebo_ros_pkgs/blob/851babe02ab5d9d758c9f129ef273bad71e5ea9c/gazebo_ros/include/gazebo_ros/node.hpp#L90

Reply:
I have added this now.